cmd/cli/kubectl-kyverno/commands/test/test.go

```diff
@@ -117,55 +117,54 @@ func runTest(openApiManager openapi.Manager, testCase test.TestCase, auditWarn b
 	var engineResponses []engineapi.EngineResponse
 	var resultCounts processor.ResultCounts
 	// TODO loop through resources first, then through policies second
-	for _, policy := range policies {
+	for _, pol := range policies {
 		// TODO we should return this info to the caller
-		_, err := policyvalidation.Validate(policy, nil, nil, true, openApiManager, config.KyvernoUserName(config.KyvernoServiceAccountName()))
+		_, err := policyvalidation.Validate(pol, nil, nil, true, openApiManager, config.KyvernoUserName(config.KyvernoServiceAccountName()))
 		if err != nil {
-			log.Log.Error(err, "skipping invalid policy", "name", policy.GetName())
+			log.Log.Error(err, "skipping invalid policy", "name", pol.GetName())
 			continue
 		}
+		matches, err := policy.ExtractVariables(pol)
+		if err != nil {
+			log.Log.Error(err, "skipping invalid policy", "name", pol.GetName())
+			continue
+		}
+		if !vars.HasVariables() && variables.NeedsVariables(matches...) {
+			// check policy in variable file
+			if !vars.HasPolicyVariables(pol.GetName()) {
+				fmt.Printf("test skipped for policy %v (as required variables are not provided by the users) \n \n", pol.GetName())
+				// TODO continue ? return error ?
```

Comment:
Return the error IMO.

Reply:
I marked it with a TODO, i think the old code didn't return...